### DISCUSSION TITLE
Continuous recording

### DISCUSSION THREAD

**Discussion by swex** [Motion-Project/motion#1492] [General]
0. Reviewed guide and contributing documents? (Yes/No): yes
1. current master
2. git
3. standalone or part of third party [motion, MotionEyeOS, other]:  motion
4. video stream source [V4L (card or USB), net cam (mjpeg, rtsp, other), mmal]: NA
5. hardware [x86, ARM, other]: NA
6. operating system [16.04, Stretch, etc, FreeBSD, other]: NA

Motion events and recordings are good, but most surveillance solutions are also provides a way to continuous recordings.
It would be great to have a way to break output videos in some customizable segments and have like motion event chapters inside(as in mkv container).
There was an old issue: #278 but it turns out it was motioneyeos related and closed after.

**Comment by usernamepi** [comment]
FYI:
The current workaround I use for for this is to use the timelapse features in the config files.
So, configure both motion capture and timelapse for the same device.
Disadvantages compared to the requested solution:
- the best fps for timelapse videos you can get is 1 fps
- limited codec/container support for timelapse videos
All motion video configuration capabilties stay the same (including fps).

**Reply by aedeny** [reply]
I do the same, but the max 1 FPS limit is a major drawback for me.

**Comment by MortenVinding** [comment]
I have been looking for a solution like that for some time.

With the current price on harddrives, I just don’t think it makes sense to only record motion events anymore.

What I want to do is stream all my cameras to my cloud server, and either let motion on my server add index points when it detects motion, or maybe even just use the build-in motion detection in the cameras to trigger index generation, for minimum processor load and power consumption.

Currently I have build my own simple system usen ffmpeg to generate .mp4 segments of 10min lenght, and a simple REST call from the camera to trigger a index pointing (URL on a list) to the segment with motion.

But I would like to use some more polished solution.

**Comment by alejsanc** [comment]
Being able to do continuous recording would be very good. With the "emulate_motion" parameter you can make it record continuously but it does not detect movements. So what I do is [run two Motion processes, one for motion detection and one for continuous recording](https://www.cuadernoinformatica.com/2024/07/grabacion-continua-de-video-con-motion.html). From one process the video is sent to a virtual device so that the other process can access it.

```
# motion.conf

movie_output off
video_pipe /dev/video1
```

```
# motion-continuous.conf

log_file /var/log/motion/motion-continuous.log
video_device /dev/video1
emulate_motion on
picture_output off
movie_max_time 1800
movie_filename continuous-%t-%v-%Y%m%d%H%M%S
```

The ideal would be to have a parameter to record continuously and detect movements at the same time.

`movie_output_continuous on`

**Comment by zarapierce01** [comment]
Currently, my workaround involves configuring both motion capture and timelapse settings for the same device in the configuration files. However, there are some drawbacks to this approach:

The maximum frame rate for timelapse videos is limited to 1 fps.

Timelapse videos have limited codec/container support.

All motion video configuration capabilities, including frame rate, remain unchanged.

For a more in-depth understanding of motion video production and its potential benefits, check out this helpful blog: [What Is Motion Video Production
.](https://www.wildcamel.tv/blog-post/what-is-motion-video-production)
